pymongo/mongo_client.py

```diff
@@ -1961,6 +1961,9 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
 
+    def __del__(self):
+        self.close()
```

Comment:
We definitely cannot call `close()` in `__del__` because the `close()` method will acquire locks, which can lead to deadlocking the garbage collector. A very similar deadlock situation is described in great detail in https://jira.mongodb.org/browse/PYTHON-799.